webui/react/src/hooks/useModal/useModalProjectCreate.tsx

```diff
@@ -0,0 +1,94 @@
+import { Input } from 'antd';
+import { ModalFuncProps } from 'antd/es/modal/Modal';
+import React, { ChangeEvent, useCallback, useEffect, useMemo, useState } from 'react';
+
+import useModal, { ModalHooks } from 'hooks/useModal/useModal';
+import { paths, routeToReactUrl } from 'routes/utils';
+import { createProject } from 'services/api';
+import handleError, { ErrorLevel, ErrorType } from 'utils/error';
+
+import css from './useModalProjectCreate.module.scss';
+
+interface Props {
+  onClose?: () => void;
+  workspaceId: number;
+}
+
+const useModalProjectCreate = ({ onClose, workspaceId }: Props): ModalHooks => {
+  const [ name, setName ] = useState('');
+  const [ description, setDescription ] = useState('');
+
+  const handleClose = useCallback(() => {
+    onClose?.();
+  }, [ onClose ]);
+
+  const { modalClose, modalOpen: openOrUpdate, modalRef } = useModal({ onClose: handleClose });
+
+  const handleNameInput = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+  }, []);
+
+  const handleDescriptionInput = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setDescription(e.target.value);
+  }, []);
+
+  const modalContent = useMemo(() => {
+    return (
+      <div className={css.base}>
+        <div>
+          <label className={css.label} htmlFor="name">Name</label>
+          <Input id="name" value={name} onChange={handleNameInput} />
+        </div>
+        <div>
+          <label className={css.label} htmlFor="description">Description</label>
+          <Input id="description" value={description} onChange={handleDescriptionInput} />
+        </div>
+      </div>
+    );
+  }, [ description, handleDescriptionInput, handleNameInput, name ]);
+
+  const handleOk = useCallback(async () => {
+    try {
+      const response = await createProject({ name, workspaceId });
+      routeToReactUrl(paths.projectDetails(response.id));
```

Comment:
for some reason `createProject` succeeds with a 200 response, but then `createProject` or `routeToReactUrl` fails. For comparison, `createWorkspace` succeeds and directs me to its page.

Reply:
Thanks, fixed!